src/requests.rs

```diff
@@ -99,28 +111,41 @@ pub async fn fetch_l1_commit_data(
     let mut prev_batch_commitment = H256::default();
     let mut curr_batch_commitment = H256::default();
     for b_number in [previous_batch_number, batch_number] {
-        let (commit_tx, _) = fetch_batch_commit_tx(b_number, network)
+        let commit_tx = fetch_batch_commit_tx(b_number, network)
             .await
-            .map_err(|e| format!("failed to find commit transaction for block {} for error: {}", b_number, e))
-            .unwrap();
+            .map_err(|_| StatusCode::FailedToFindCommitTxn);
```

Comment:
strange - first you're mapping the error to FailedToFind - but then 2 lines below, you ignore the contents and return the hardcoded error anywaysl

Reply:
good catch, updating